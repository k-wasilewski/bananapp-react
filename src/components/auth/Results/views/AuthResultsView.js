import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class AuthResultsView extends Component {
    render() {
        return (
            <div className='App'>
                <header className='App-header'>
                    <img src={this.props.img} className='img' alt='your banana' />
                    <h3> Your banana is { this.props.days } old, </h3>
                    <h3> with { Number((this.props.accuracy*100).toFixed(2)) }%
                        certainty</h3>
                    <Link to='/success'>
                        <button variant='outlined'>
                            Back
                        </button>
                    </Link>
                </header>
            </div>
        )
    }
}
export default AuthResultsView;