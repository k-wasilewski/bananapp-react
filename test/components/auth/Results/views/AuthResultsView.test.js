import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import AuthResultsView from "../../../../../src/components/auth/Results/views/AuthResultsView";

describe("AuthResultsView rendering specification", () => {
    it('AuthResultsView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <AuthResultsView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});