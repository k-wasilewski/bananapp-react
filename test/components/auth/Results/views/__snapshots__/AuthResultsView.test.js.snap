// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthResultsView rendering specification AuthResultsView is rendered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h3>
      File type not supported or it's not a banana, try again
    </h3>
    <a
      href="/success"
      onClick={[Function]}
    >
      <button
        variant="outlined"
      >
        Back
      </button>
    </a>
  </header>
</div>
`;
